src/crypto/cryptobox.h

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+extern "C"
+{
+#include <evercrypt/EverCrypt_Curve25519.h>
+#include <evercrypt/Hacl_NaCl.h>
+}
+
+#include <fmt/format_header_only.h>
+#include <vector>
+
+namespace crypto
+{
+  class BoxKey
+  {
+  public:
+    static constexpr size_t KEY_SIZE = 32;
+
+    static std::vector<uint8_t> public_from_private(
+      std::vector<uint8_t>& private_key)
+    {
+      if (private_key.size() != KEY_SIZE)
+      {
+        throw std::logic_error(
+          fmt::format("Private key size is not {}", KEY_SIZE));
```

Comment:
would it be helpful here in all the errors where the key size is not correct to also print the key size that we got?